tasks/utils/i18n_verify_keys.js

```diff
@@ -0,0 +1,58 @@
+import fs from 'fs';
+import path from 'path';
+import Promise from 'bluebird';
+import globProm from 'glob-promise';
+
+const readFile = Promise.promisify(fs.readFile);
+
+/**
+ * Verify translation keys in files are translated.
+ * @param {Array<string>} filesPatterns - List of file patterns to be checkd for translation keys
+ * @param {Object}} translations - Object of translations keys and their translations
+ * @param {RegExp} translationPatternRegEx - Pattern of translation method to check for so can get list of translation keys
+ * @return {Promise} - A Promise object which will contain an empty Object if all translation keys are translated. If translation keys are
+ * not translated then the Object will contain all non translated translation keys with value of file the key is from
+ */
+export function verifyTranslationKeys(filesPatterns, translations, translationPatternRegEx) {
+  return getFilesToVerify(filesPatterns).then(function (filesToVerify) {
+    return verifyKeysInFiles(filesToVerify, translationPatternRegEx, translations).then(function (keysNotTranslated) {
+      return keysNotTranslated;
+    });
+  });
+};
+
+function getFilesToVerify(verifyFilesPatterns) {
+  let filesToVerify = [];
+
+  return Promise.map(verifyFilesPatterns, (verifyFilesPattern) => {
+    const baseSearchDir = path.dirname(verifyFilesPattern);
+    const pattern = path.basename(verifyFilesPattern);
+    return globProm(pattern, {cwd: baseSearchDir, matchBase: true}).then(function (files) {
+      for (let file of files) {
+        filesToVerify.push(baseSearchDir + '/' + file);
+      }
+    });
+  }).then(function () {
+    return filesToVerify;
+  });
+}
+
+function verifyKeysInFiles(filesToVerify, translationPatternRegEx, translations) {
+  let keysNotTranslated = {};
+
+  return Promise.all(filesToVerify, (file) => {
```

Comment:
This isn't quite what you want. Promise.all takes a single argument which is an iterable containing promises. You have to build the promises first. So you'll want to do something like:
```
const filePromises = _.map(filesToVerify, (file) => {
  return readFile(file, 'utf8')
  .then((fileContents) => {
    // do more stuff
  });
});
return Promise.all(filePromises);
```